lightning-c-bindings/include/lightning.h

```diff
@@ -0,0 +1,5356 @@
+   bool _underlying_ref;
+} LDKLocalCommitmentTransaction;
+
+
+
+/**
+ *  The unsigned part of a channel_announcement
+ */
+typedef struct MUST_USE_STRUCT LDKUnsignedChannelAnnouncement {
+   /**
+    * Nearly everyhwere, inner must be non-null, however in places where
+    * the Rust equivalent takes an Option, it may be set to null to indicate None.
+    */
+   LDKlnUnsignedChannelAnnouncement *inner;
+   bool _underlying_ref;
+} LDKUnsignedChannelAnnouncement;
+
+/**
+ *  Set of lightning keys needed to operate a channel as described in BOLT 3.
+ *
+ *  Signing services could be implemented on a hardware wallet. In this case,
+ *  the current ChannelKeys would be a front-end on top of a communication
+ *  channel connected to your secure device and lightning key material wouldn't
+ *  reside on a hot server. Nevertheless, a this deployment would still need
+ *  to trust the ChannelManager to avoid loss of funds as this latest component
+ *  could ask to sign commitment transaction with HTLCs paying to attacker pubkeys.
+ *
+ *  A more secure iteration would be to use hashlock (or payment points) to pair
+ *  invoice/incoming HTLCs with outgoing HTLCs to implement a no-trust-ChannelManager
+ *  at the price of more state and computation on the hardware wallet side. In the future,
+ *  we are looking forward to design such interface.
+ *
+ *  In any case, ChannelMonitor or fallback watchtowers are always going to be trusted
+ *  to act, as liveness and breach reply correctness are always going to be hard requirements
+ *  of LN security model, orthogonal of key management issues.
+ *
+ *  If you're implementing a custom signer, you almost certainly want to implement
+ *  Readable/Writable to serialize out a unique reference to this set of keys so
+ *  that you can serialize the full ChannelManager object.
+ *
+ */
+typedef struct LDKChannelKeys {
+   void *this_arg;
+   /**
+    *  Gets the per-commitment point for a specific commitment number
+    *
+    *  Note that the commitment number starts at (1 << 48) - 1 and counts backwards.
+    */
+   LDKPublicKey (*get_per_commitment_point)(const void *this_arg, uint64_t idx);
+   /**
+    *  Gets the commitment secret for a specific commitment number as part of the revocation process
+    *
+    *  An external signer implementation should error here if the commitment was already signed
+    *  and should refuse to sign it in the future.
+    *
+    *  May be called more than once for the same index.
+    *
+    *  Note that the commitment number starts at (1 << 48) - 1 and counts backwards.
+    *  TODO: return a Result so we can signal a validation error
+    */
+   LDKThirtyTwoBytes (*release_commitment_secret)(const void *this_arg, uint64_t idx);
+   /**
+    *  Gets the local channel public keys and basepoints
+    */
+   LDKChannelPublicKeys pubkeys;
+   /**
+    * Fill in the pubkeys field as a reference to it will be given to Rust after this returns
```

Comment:
Should this be part of the user documentation? It seems like an implementation detail, but I see it called in `demo.cpp`. If it should stay, I think it may need some clarification.

Reply:
Its in a standard documentation comment? Do you mean you'd like to see a note about this specific function added to the README?

Reply:
Let me try to better explain why I'm confused. The original `ChannelKeys` rust trait has a `pubkeys` method that implementors of the trait must define. But when reading the demo code, I see the user needs to explicitly call `set_pubkeys` on a `LDK::ChannelKeys` that is returned by an `LDK::KeysInterface` function:

https://github.com/rust-bitcoin/rust-lightning/blob/7e06d084622f0c2e76e74981f72fb15257ea4e7e/lightning-c-bindings/demo.cpp#L245

However, if someone is working with Rust directly, they presumably wouldn't need to do anything like this. So what is it about the bindings that requires the user to explicitly set this?

Reply:
I further updated the README to hopefully contain less extraneous detail and also capture what is going on.